### PR TITLE
fix(admin-api) incorrect PUT behavior for /certificates

### LINKED ISSUE #5309: bug: incorrect PUT behavior for /certificates
### Summary

PUT behavior on `/certificate` doesn't reset `snis`.

### Steps To Reproduce

1. `http :8001/certificates cert=@cert.pem key=@key.pem 'snis[]=foo.com' id=41e72c04-3172-4336-b4ce-fefa2a321679 -f`
2. `http put :8001/certificates/41e72c04-3172-4336-b4ce-fefa2a321679 cert=@cert.pem key=@key.pem -f`
3. `snis` is still set to `[ foo.com ]`

### Additional Details & Logs

- Kong version (`$ kong version`) 1.4.1 (doesn't matter, this probably is the case since 1.0)
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (the output of a GET request to Kong's Admin port - see
  https://docs.konghq.com/latest/admin-api/#retrieve-node-information)
- Operating system Docker

This is due to the "sugar" API that we have on top of two DAO: `certificate` and `sni`.

### PR DESCRIPTION
**Pull request by bungle**
### Summary

@hbagdi reported issue on `PUT` behavior on `/certificates` with #5309.

This PR fixes the `PUT` behavior. It also fixes several other issues
found while fixing the original issue.

### Issues Resolver

Fix #5309

### CODE REVIEW

File: spec/02-integration/04-admin_api/06-certificates_routes_spec.lua [hunk 1]
```diff
@@ -440,6 +440,29 @@ describe("Admin API: #" .. strategy, function()
...
+      it_content_types("update by id returns full certificate", function(content_type)
+        return function()
+          local body
+          if content_type == "application/x-www-form-urlencoded" then
+            body = {
+            }
+          else
+            body = {
+            }
+          end
```
Comment (hbagdi) [inline spec/02-integration/04-admin_api/06-certificates_routes_spec.lua]: Question: what purpose does these `if` block serve? :thinking:

Comment (bungle) [reply spec/02-integration/04-admin_api/06-certificates_routes_spec.lua]: haha, :-) just copy pasting...

File: spec/02-integration/04-admin_api/06-certificates_routes_spec.lua [hunk 2]
```diff
@@ -326,7 +326,7 @@ describe("Admin API: #" .. strategy, function()
         assert.same(json, in_db)
       end)
 
-      it("updates if found", function()
+      it("upserts if found", function()
```
Comment (hutchic) [inline spec/02-integration/04-admin_api/06-certificates_routes_spec.lua]: TIL this is a thing

**Comment by hishamhm** [comment]
@hutchic I see it as a fix, since that is the overall behavior that we advertise for entities under `PUT` in our documentation.

**Comment by hbagdi** [comment]
I'm not fluent enough with this part to confidently review the change but the test changes look correct.
I think this is a `fix` and not a `feat` too.

**Pull request merged**